### PR TITLE
feat: integrate zap logger and datadog statsd

### PR DESCRIPTION
**Pull request by GeekArthur**
Integrate zap logger and datadog statsd

### CODE REVIEW

File: types/errors.go [hunk 1]
```diff
@@ -164,28 +169,28 @@ var (
 
 	// ErrNonceError is returned when we are unable to get nonce
 	ErrNonceError = &types.Error{
-		Code:    15, //nolint
+		Code:    16, //nolint
```
Comment (shrimalmadhur) [inline types/errors.go:172]: lol good catch

File: utils/bootstrap.go [hunk 2]
```diff
@@ -45,6 +45,23 @@ func BootStrap(
...
+		return err
+	}
+
+	logger, syncFn, err := stats.InitLogger(cfg)
+	if err != nil {
+		return err
+	}
+	defer syncFn()
+
+	statsdClient, done, err := stats.InitStatsd(logger, cfg, "c3/rosetta-api")
```
Comment (shrimalmadhur) [inline utils/bootstrap.go:59]: we might want to make this as env variable too.

Comment (GeekArthur) [reply utils/bootstrap.go:59]: Yup, will update this

Comment (GeekArthur) [reply utils/bootstrap.go]: Done

**Review by shrimalmadhur — APPROVED** [review]

**Pull request closed without merging**